syntax ="proto3";

service PointRegistering{
    rpc registerPoints(input) returns (output){}
}
message input{
    repeated vector point_set_1=1;
    repeated vector point_set_2=2;
    string algorithm=3;
}
message output{
    string status=1;
    repeated matrix_row rotation_matrix=2;
    vector translation_vector=3;
   
}
message matrix_row{
    repeated float row=1;
}
message vector{
    repeated float vector_data=1;
}